contracts/utils/math/Math.sol

```diff
@@ -335,6 +330,59 @@ library Math {
         }
     }
 
+    /**
+     * @dev Variant of {modExp} that supports inputs of arbitrary length.
+     */
+    function modExp(bytes memory b, bytes memory e, bytes memory m) internal view returns (bytes memory) {
+        (bool success, bytes memory result) = tryModExp(b, e, m);
+        if (!success) {
+            Panic.panic(Panic.DIVISION_BY_ZERO);
+        }
+        return result;
+    }
+
+    /**
+     * @dev Variant of {tryModExp} that supports inputs of arbitrary length.
+     */
+    function tryModExp(
+        bytes memory b,
+        bytes memory e,
+        bytes memory m
+    ) internal view returns (bool success, bytes memory result) {
+        if (_zeroBytes(m)) return (false, result);
```

Comment:
Well, `a / 0 == undefined`.

Let's keep `new bytes(0)`, but imo the reason why we shouldn't care about reallocating the memory pointer if we're returning `new bytes(0)` should be because the memory leak is minimal (just the size slot) and not because the memory may be dirty.

Reply:
Technically, is you want to do `x = new bytes(0)`, you don't even need to allocate new data. It would be enough to point to 0x60 (that should always contain zero).

Its a memory array, so you can't push anyway.